zrml/simple-disputes/src/lib.rs

```diff
@@ -31,68 +31,195 @@ pub use simple_disputes_pallet_api::SimpleDisputesPalletApi;
+        #[pallet::constant]
+        type DisputeBond: Get<BalanceOf<Self>>;
+
+        /// The additional amount of currency that must be bonded when creating a subsequent
+        /// dispute.
+        #[pallet::constant]
+        type DisputeFactor: Get<BalanceOf<Self>>;
+
         /// The identifier of individual markets.
         type MarketCommons: MarketCommonsPalletApi<
             AccountId = Self::AccountId,
             BlockNumber = Self::BlockNumber,
         >;
 
+        /// The maximum number of disputes allowed on any single market.
+        #[pallet::constant]
+        type MaxDisputes: Get<u32>;
+
         /// The pallet identifier.
         #[pallet::constant]
         type PalletId: Get<PalletId>;
+
+        #[pallet::constant]
+        type PredictionMarketsPalletId: Get<PalletId>;
     }
 
+    type BalanceOf<T> =
+    <CurrencyOf<T> as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+    pub(crate) type CurrencyOf<T> =
+        <<T as Config>::MarketCommons as MarketCommonsPalletApi>::Currency;
+    pub(crate) type MarketIdOf<T> =
+        <<T as Config>::MarketCommons as MarketCommonsPalletApi>::MarketId;
+    pub(crate) type MomentOf<T> = <<T as Config>::MarketCommons as MarketCommonsPalletApi>::Moment;
+    pub(crate) type MarketOf<T> = Market<
+        <T as frame_system::Config>::AccountId,
+        BalanceOf<T>,
+        <T as frame_system::Config>::BlockNumber,
+        MomentOf<T>,
+    >;
+
+    #[pallet::pallet]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    /// For each market, this holds the dispute information for each dispute that's
+    /// been issued.
+    #[pallet::storage]
+    pub type Disputes<T: Config> = StorageMap<
+        _,
+        Blake2_128Concat,
+        MarketIdOf<T>,
+        BoundedVec<MarketDispute<T::AccountId, T::BlockNumber>, T::MaxDisputes>,
+        ValueQuery,
+    >;
+
+    #[pallet::event]
+    #[pallet::generate_deposit(fn deposit_event)]
+    pub enum Event<T>
+    where
+        T: Config, {
+            OutcomeReserved { market_id: MarketIdOf<T>, dispute: MarketDispute<T::AccountId, T::BlockNumber> },
+        }
+
     #[pallet::error]
     pub enum Error<T> {
         /// 1. Any resolution must either have a `Disputed` or `Reported` market status
         /// 2. If status is `Disputed`, then at least one dispute must exist
         InvalidMarketStatus,
         /// On dispute or resolution, someone tried to pass a non-simple-disputes market type
         MarketDoesNotHaveSimpleDisputesMechanism,
+        StorageOverflow,
+        OutcomeMismatch,
+        CannotDisputeSameOutcome,
+        MarketIsNotReported,
+        MaxDisputesReached,
     }
 
-    #[pallet::event]
-    pub enum Event<T>
-    where
-        T: Config, {}
-
     #[pallet::hooks]
     impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {}
 
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::weight(5000)]
+        #[transactional]
+        pub fn reserve_outcome(
```

Comment:
block this extrinsic for main-net

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/938/commits/f1347c3eaaf971af9c224913f19091c8ac9426c6